src/vmm/src/pipe.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright 2020 Sartura All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+use std::fs::{File, OpenOptions};
+use std::io::{Read, Write};
+use std::os::unix::io::{AsRawFd, RawFd};
+use std::path::PathBuf;
+
+/// A wrapper structure around that provides basic read functionality backed by
+/// a named pipe implementation.
+pub(crate) struct PipeReaderWrapper(PathBuf, File);
+
+impl PipeReaderWrapper {
+    pub fn new(buf: PathBuf) -> PipeReaderWrapper {
+        match utils::mkfifo(&buf, libc::S_IRWXO) {
+            Ok(_) => (),
+            Err(err) => panic!("Error creating fifo: {}", err),
```

Comment:
We should return an error instead of panic. Same for all the panics in this file.

Once we define an error, we should also use `map_err` and convert these lines to 1-liners.